### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by maximshenets**
Authentication implemented.

### CODE REVIEW

File: app/config/run.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export default function run($rootScope, $location, AUTH_EVENTS, authService) {
+  $rootScope.$on('$stateChangeStart', function(event, next) {
+    var isAuthenticated = next.data.isAuthenticated;
+    if(isAuthenticated) {
+      if(authService.isAuthenticated()) {
+        console.log(authService.isAuthenticated());
+      } else {
+        event.preventDefault();
+        $location.path('/login');
+        $rootScope.$broadcast(AUTH_EVENTS.notAuthenticated)
```
Comment (alekseyC64) [inline app/config/run.js]: what is the purpose of this event?

File: debug.log [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[0221/154155:ERROR:tcp_listen_socket.cc(76)] Could not bind socket to 127.0.0.1:6004
```
Comment (alekseyC64) [inline debug.log]: should be git ignored, I suppose

File: app/config/routes.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+export default function routes($stateProvider, $urlRouterProvider) {
+
+  $urlRouterProvider.otherwise('/');
+
+  $stateProvider
```
Comment (NikitaSedyx) [inline app/config/routes.js]: nit:
```
$stateProvider
  .state
```
please

File: app/config/run.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export default function run($rootScope, $location, AUTH_EVENTS, authService) {
```
Comment (NikitaSedyx) [inline app/config/run.js]: imo it's better to move `export` to the end of the file

File: app/controllers/application.controller.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export default function ApplicationController($scope, AUTH_EVENTS, authService) {
+  $scope.currentUser = null;
+
+  $scope.setCurrentUser = function(user) {
+    $scope.currentUser = user;
+  }
```
Comment (NikitaSedyx) [inline app/controllers/application.controller.js]: nit: line please

File: app/controllers/application.controller.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  $scope.logout = function() {
+    authService.logout();
+    $scope.setCurrentUser(null);
+    $scope.$broadcast(AUTH_EVENTS.logoutSuccess);
```
Comment (NikitaSedyx) [inline app/controllers/application.controller.js]: `$scope.$broadcast`?

File: app/controllers/login.controller.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    password: ''
+  };
+  $scope.login = function(credentials) {
+    authService.login(credentials)
+    .then(
+    function(user) {
+      $rootScope.$broadcast(AUTH_EVENTS.loginSuccess);
+      $scope.setCurrentUser(user);
+    },
+    function() {
```
Comment (NikitaSedyx) [inline app/controllers/login.controller.js]: `.catch`?

File: app/controllers/login.controller.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      $rootScope.$broadcast(AUTH_EVENTS.loginFailed);
+    });
+  };
+  // $scope.login = function() {
```
Comment (NikitaSedyx) [inline app/controllers/login.controller.js]: is this code required?

File: app/services/auth.service.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  function login(credentials) {
+    return $http
+      .post('api/v1/auth/login/', credentials)
+      .then(function(response) {
+        Session.create(response.data.session_id, response.data.user.id);
+        console.log(response);
+        return response.data.user;
+      })
+  }
+  // function login() {
```
Comment (NikitaSedyx) [inline app/services/auth.service.js]: same

**Pull request merged**